components/filter/filter-dimension-set-date-text-value.js

```diff
@@ -0,0 +1,149 @@
+import { getUTCDateTimeRange } from '../../helpers/dateTime.js';
+import { LitElement } from 'lit';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+
+/**
+ * A component to represent a possible date value that can be selected for a dimension set (the main filter dimension type) for predefined date ranges.
+ * A range property is used to define the preset text shown, as well as the start and end date values formatted as UTC strings.
+ * This component does not render anything, but instead gathers data needed for the d2l-filter.
+ */
+class FilterDimensionSetDateTextValue extends LocalizeCoreElement(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether this value in the filter is disabled or not
+			 * @type {boolean}
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * @ignore
+			 */
+			endValue: { type: String },
+			/**
+			 * REQUIRED: Unique key to represent this value in the dimension
+			 * @type {string}
+			 */
+			key: { type: String },
```

Comment:
Does the key serve any purpose to us or the consumer if they are using `rangeNum` & `rangeType`?